### ISSUE TITLE
Dec 1-2: Predict human activity with random forest classification

### ISSUE THREAD

**Issue by meaghanlewis** [microsoft/ReactorPlanning#123]
# Event Information
| Category | Details |
|-----------|---------|
| Reactor Topic |  Data Science and Machine Learning | 
| Workshop Duration | 60 min | 
| Title | Random Forest Classification  |
| Description | In this workshop we will learn about and create a model using random forest classification. With random forest, decision trees are created based on data samples and then predictions are made about which class they belong to. We'll use data collected by smartphone sensors to detect what activity a person is performing, from walking, sitting, standing, or laying down. This workshop is for people who have a basic understanding of Machine Learning Fundamentals. To follow along, you'll need Visual Studio Code and Python installed. Previous experience with coding in Python is recommended. |
| Content| https://aka.ms/DSHumanBehaviorProject | 

# Workshop(s) details
| Date |  Local Start Time |  Location |  Speaker | SharePoint ID | Platform |
|------|-------------------|-----------|----------|---------------|----------|
| 12/1 | 8:30 PM | Bangalore | Michael Galarnyk | 11884 |----------|
| 12/1 | 4:00 PM | US, Canada, Sydney | Michael Galarnyk  | 11885 |----------|
| 12/2 | 7:30 PM | Sao Paulo | Fabricio Murai Ferreira | 11886 |----------|
| 12/2 | 7:30 PM | Shanghai | Ryan Chung | 11888 |----------|
| 12/2 | 12:00 PM | London, Stockholm, Abu Dhabi, Tel Aviv | Brian Sletten | 11887 |----------|

# Event Checklist

## 10 Weeks Prior
- [x] Reactor Content Team (RCT) opens a new issue for an upcoming local workshop filling in the Event Information and Workshop details tables. RCT member who owns content (content owner) this week will be the assignee.
- [x] RCT and content owner discuss details of the event in the comments of this issue. RCT adds a _confirmed-event_ label and _local-digital-event_ tag to the issue and adds event to the __Local Digital Events__ project. Issue will appear in the __Upcoming Streams__ column
- [x] RCT adds a _ready-for-marketing-edit_ label so the RCT can review title and description

## 8 Weeks Prior
- [x] RCT does a review of title and description and removes the _ready-for-marketing-edit_ label 
- [x] RCT moves the issue to the __Ready for SharePoint__ column
- [ ] RCT informs the Reactor Marketing Team of the new event information
- [ ] RCT confirms the speaker
- [ ] RCT provides speaker with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT moves the issue to the __Content Preparation In Progress__ column

## 5 weeks Prior
- [ ] Content owner opens a PR with the new content for the workshop on the [customer-facing Reactor repository](https://github.com/microsoft/Reactors/tree/main/online-event-resources) and adds the _content-development_ and _local-digital-event_ tag to the PR. 
- [ ] RCT reviews, approves, and merges the changes
- [ ] RCT shares content with the vendor/instructor
- [ ] RCT provides vendor/instructor with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT ensures that instructors have received an invite to the workshops and that PM team works with instructors to get setup in the platform (e.g. Teams, Teams Live Event)

## 1 Week Prior
- [ ] Connect vendor/instructor with the PM(s) for the locations where this event has been marketed
- [ ] Check-in with vendor/instructor that they are prepared for the workshop
- [ ] RCT moves the issue to the __Ready to Stream__ column

## During and After Stream
- [ ] Instructor (and optional moderator) run the stream
- [ ] _(Optional)_RCT submits recording to be uploaded to the Reactor YouTube and to be made available on LearnTV
- [ ] RCT moves issue to the __Stream Complete__ column

**Comment by mebaumb** [comment]
Hi @meaghanlewis - what do you think? 

In this workshop we will learn about and create a model using random forest classification. With random forest, decision trees are created based on data samples and then predictions are made about which class they belong to. We'll use data collected by smartphone sensors to detect what activity a person is performing, from walking, sitting, standing, or laying down. This workshop is for people who have a basic understanding of Machine Learning Fundamentals.

To follow along, you'll need Visual Studio Code and Python installed. 

Previous experience with coding in Python is recommended.

**Comment by meaghanlewis** [comment]
Hey @mebaumb this looks good to me. This is not from MS Learn so I don't think we need to include any links in the description to the content. I'll just specify in the instructor notes the aka link they can share during the workshop.

I made a couple edits inline above.

**Comment by meaghanlewis** [comment]
Here's an overview of what this workshop will cover and suggested agenda to follow with links and resources.

## Content to teach
This workshop will take Reactor community members through the [Data Science 2 Human Behavior Project](https://aka.ms/DSHumanBehaviorProject)

**SHARE THIS LINK WITH REACTOR ATTENDEES: https://aka.ms/DSHumanBehaviorProject**

## Agenda
1. [Explain the purpose of the workshop](https://github.com/microsoft/Reactors/blob/main/workshop-resources/data-science-and-machine-learning/Data_Science_2/human-behavior-project/2-behavior-activity.md) 5-10 min
   - Explain the problem that we're trying to solve in the workshop
   - Show the [behavior study](https://github.com/microsoft/Reactors/blob/main/workshop-resources/data-science-and-machine-learning/Data_Science_2/human-behavior-project/behavior_study.md) and explain what happened in the study
   - Show the [test and train data sets](https://github.com/microsoft/Reactors/tree/main/workshop-resources/data-science-and-machine-learning/Data_Science_2/human-behavior-project/Data)
   - Explain how they can follow along during the workshop
      - **Visual Studio Code** is needed
      - Python extension is needed
      - Familiarity with Python recommended
      - Encourage attendees to download the data and .ipynb file needed
2. [Walk through the notebook](https://github.com/microsoft/Reactors/blob/main/workshop-resources/data-science-and-machine-learning/Data_Science_2/human-behavior-project/human_behavior_activity.ipynb) 45 min
    - Make sure everyone can get their Jupyter Notebook running in VS Code
    - Go through each section explaining what's happening
    - Encourage questions and answer them as they come in
3. **Next steps** 2 min
    - Encourage attendees to reference this notebook and go through it on their own time as needed
4. **Closing** 5 min
    - Q&A
    - **Ask attendees to take the survey: https://aka.ms/Reactor/Survey**
      - Note: The event code is listed in this issue description and/or your Reactor contact can share that with the community members during the workshop
    - Share Resources
      - Encourage attendees to view other DS and ML workshop materials here: https://github.com/microsoft/Reactors/tree/main/workshop-resources/data-science-and-machine-learning
    - Check if the Reactor contact has other information to share

**Comment by mebaumb** [comment]
@meaghanlewis - is this event good to publish with the edited description then?

**Comment by meaghanlewis** [comment]
@mebaumb yep!

**Comment by mebaumb** [comment]
All Events except Sao Paulo have been published. Waiting on translation for Sao Paulo

**Comment by mebaumb** [comment]
Stream analytics have been added to tracking documents.
